From bb0435e3f3a934aece6ed08058ba0f767e9cbd78 Mon Sep 17 00:00:00 2001
From: yasminvalim <ydesouza@redhat.com>
Date: Fri, 5 Sep 2025 17:50:32 -0300
Subject: [PATCH] blockdev.rs: uses blkid -p instead of lsblk to bypass the
 cache

release-notes: add notes and comment for more context
(cherry picked from commit d7b39b91268406e391f8fff246d81a7160cbc5d9)
---
 src/blockdev.rs | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/blockdev.rs b/src/blockdev.rs
index 504a5a7..63d6522 100644
--- a/src/blockdev.rs
+++ b/src/blockdev.rs
@@ -494,7 +494,9 @@ impl Mount {
     }
 
     pub fn get_filesystem_uuid(&self) -> Result<String> {
-        let devinfo = lsblk_single(Path::new(&self.device))?;
+        // We used to use lsblk_single, but its cache may be stale after mkfs.
+        // blkid_single doesn't use cache.
+        let devinfo = blkid_single(Path::new(&self.device))?;
         devinfo
             .get("UUID")
             .map(String::from)
-- 
2.52.0

